### PR TITLE
OI-278 Image cropping

### PR DESCRIPTION
**Pull request by nvelychenko**

### CODE REVIEW

File: modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Provides base class for openidealEntityStatistics blocks.
+ */
+abstract class OpenidealStatisticsBaseStatisticsBlock extends BlockBase {
+
+  /**
+   * Checks if it's a full view mode.
```
Comment (db-Matroskeen) [inline modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php]: The description doesn't match.

File: modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+   *
+   * @return bool
+   *   True is not provided view mode.
+   */
+  public function isNotViewMode($mode) {
```
Comment (db-Matroskeen) [inline modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php]: Wouldn't it better to make it `isViewMode()` or `getViewMode()`?

File: modules/openideal_user/openideal_user.install [hunk 3]
```diff
@@ -14,3 +14,26 @@ function openideal_user_update_8001() {
...
+function openideal_user_update_8002() {
+  /** @var \Drupal\update_helper\Updater $updateHelper */
+  $updateHelper = \Drupal::service('update_helper.updater');
+
+  // Execute configuration update definitions with logging of success.
+  $updateHelper->executeUpdate('openideal_user', 'openideal_user_update_8002');
+
+  // Change image style for user author view mode picture.
+  $config = \Drupal::configFactory()->getEditable('core.entity_view_display.user.user.author');
+  $path = 'third_party_settings.layout_builder.sections.0.components.905abf77-399a-4a9b-826a-2cf21309bbe5.configuration.formatter.settings.image_style';
```
Comment (db-Matroskeen) [inline modules/openideal_user/openideal_user.install:30]: Is UUID the same for everyone?

File: modules/openideal_media/openideal_media.info.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+name: 'OpenideaL Media'
```
Comment (db-Matroskeen) [inline modules/openideal_media/openideal_media.info.yml:1]: Where do we enable this module during the update?

**Review by db-Matroskeen — CHANGES_REQUESTED** [review]

File: modules/openideal_media/openideal_media.info.yml (hunk 4 above)
Comment (nvelychenko) [reply modules/openideal_media/openideal_media.info.yml:1]: Here https://github.com/istolar/openideal/pull/54/files#diff-e42e8e09ccc374119c0a3d6217ad49e5cb71fc929a5c6aacedfd9ec490c61c66R6

File: modules/openideal_user/openideal_user.install (hunk 3 above)
Comment (nvelychenko) [reply modules/openideal_user/openideal_user.install:30]: Yes

File: modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php (hunk 2 above)
Comment (nvelychenko) [reply modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php]: We are checking if the view provided as method argument not active, and return a bool value.
For instance, if the method called getViewMode then it should return the actual name of the view mode.

File: modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php (hunk 1 above)
Comment (nvelychenko) [reply modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php]: Yep, changed. Thanks.

File: modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php [hunk 5]
```diff
@@ -4,20 +4,41 @@
...
   /**
    * Checks if provided view mode not active.
    *
    * @return bool
    *   True is not provided view mode.
    */
-  public function isNotViewMode($mode) {
-    return isset($this->getContexts()['view_mode']) && ($this->getContexts()['view_mode']->getContextValue() != $mode);
+  public function isViewMode($mode) {
+    // Can't check view mode of blocks that are placed as plain structure
```
Comment (db-Matroskeen) [inline modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsBaseStatisticsBlock.php]: What is "plain structure"?

**Pull request merged**